business/tls.go

```diff
@@ -59,44 +57,28 @@ func (in *TLSService) getMeshPeerAuthentications() ([]security_v1beta1.PeerAuthe
 }
 
 func (in *TLSService) getAllDestinationRules(namespaces []string) ([]networking_v1alpha3.DestinationRule, error) {
-	drChan := make(chan []networking_v1alpha3.DestinationRule, len(namespaces))
-	errChan := make(chan error, 1)
-	wg := sync.WaitGroup{}
-
-	wg.Add(len(namespaces))
-
-	for _, namespace := range namespaces {
-		go func(ns string) {
-			defer wg.Done()
-			criteria := IstioConfigCriteria{
-				Namespace:               ns,
-				IncludeDestinationRules: true,
-			}
-			istioConfigList, err := in.businessLayer.IstioConfig.GetIstioConfigList(criteria)
-			if err != nil {
-				errChan <- err
-				return
-			}
-
-			drChan <- istioConfigList.DestinationRules
-		}(namespace)
+	criteria := IstioConfigCriteria{
+		AllNamespaces:           true,
+		IncludeDestinationRules: true,
 	}
 
-	wg.Wait()
-	close(errChan)
-	close(drChan)
-
-	for err := range errChan {
-		if err != nil {
-			return nil, err
-		}
+	istioConfigList, err := in.businessLayer.IstioConfig.GetIstioConfigList(criteria)
+	if err != nil {
+		return nil, err
 	}
 
 	allDestinationRules := make([]networking_v1alpha3.DestinationRule, 0)
-	for drs := range drChan {
-		allDestinationRules = append(allDestinationRules, drs...)
+	for _, dr := range istioConfigList.DestinationRules {
+		found := false
+		for _, ns := range namespaces {
+			if dr.Namespace == ns {
+				found = true
+			}
+		}
+		if found {
+			allDestinationRules = append(allDestinationRules, dr)
```

Comment:
I think here all destination rules should be filtered by "filterDRExportToNamespaces".
Because in "WorkloadMtlsStatus" method these DRs hostis  checked to match the Service, so DR should be available for the namespace of the Service with matching selector labels.

Reply:
But this can be done in a scope of dependent https://github.com/kiali/kiali/pull/4567

Reply:
Hi @hhovsepy I'm resuming the work for this PR, I'm ok to refine this in the #4567, my goal of the work is just to update the business/tls.go and validate that testing don't show any regression due the changes on the mocking artifacts.